src/meepgeom.cpp

```diff
@@ -748,6 +795,9 @@ void geom_epsilon::eff_chi1inv_matrix(meep::component c, symmetric_matrix *chi1i
     }
   }
 
+  if (mat->which_subclass == material_data::MATERIAL_GRID) {
+    goto noavg;
+  }
```

Comment:
If instead you return `true` for `is_variable` on `MATERIAL_GRID`, then `get_front_object` will return `false` and you can use that branch instead.